streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorNodeTest.java

```diff
@@ -34,36 +40,106 @@
     }
 
     @Test
     public void shouldThrowStreamsExceptionIfExceptionCaughtDuringClose() {
         final ProcessorNode<Object, Object, Object, Object> node =
-            new ProcessorNode<>("name", new ExceptionalProcessor(), Collections.emptySet());
+            new ProcessorNode<>(NAME, new ExceptionalProcessor(), Collections.emptySet());
         assertThrows(StreamsException.class, () -> node.init(null));
     }
 
+    @Test
+    public void shouldThrowStreamsExceptionWhenProcessingMarkedAsFail() {
+        final ProcessorNode<Object, Object, Object, Object> node =
+            new ProcessorNode<>(NAME, new ProcessingExceptionProcessor(), Collections.emptySet());
+
+        final InternalProcessorContext<Object, Object> internalProcessorContext = mockInternalProcessorContext();
+        node.setProcessingExceptionHandler(new ProcessingExceptionHandlerMock(ProcessingExceptionHandler.ProcessingHandlerResponse.FAIL, internalProcessorContext));
+        node.init(internalProcessorContext);
+
+        final FailedProcessingException failedProcessingException = assertThrows(FailedProcessingException.class,
+            () -> node.process(new Record<>(KEY, VALUE, TIMESTAMP)));
+
+        assertEquals("java.lang.RuntimeException: Processing exception should be caught and handled by the processing exception handler.",
+            failedProcessingException.getMessage());
+
+        assertTrue(failedProcessingException.getCause() instanceof RuntimeException);
+        assertEquals("Processing exception should be caught and handled by the processing exception handler.",
+            failedProcessingException.getCause().getMessage());
+    }
+
+    @Test
+    public void shouldNotThrowStreamsExceptionWhenProcessingMarkedAsContinue() {
+        final ProcessorNode<Object, Object, Object, Object> node =
+            new ProcessorNode<>(NAME, new ProcessingExceptionProcessor(), Collections.emptySet());
+
+        final InternalProcessorContext<Object, Object> internalProcessorContext = mockInternalProcessorContext();
+        node.setProcessingExceptionHandler(new ProcessingExceptionHandlerMock(ProcessingExceptionHandler.ProcessingHandlerResponse.CONTINUE, internalProcessorContext));
+        node.init(internalProcessorContext);
+
+        assertDoesNotThrow(() -> node.process(new Record<>(KEY, VALUE, TIMESTAMP)));
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void shouldNotHandleInternalFailedProcessingException() {
+        final ProcessingExceptionHandler processingExceptionHandler = mock(ProcessingExceptionHandler.class);
+
+        final ProcessorNode<Object, Object, Object, Object> node =
+            new ProcessorNode<>(NAME, new FailedProcessingExceptionProcessor(), Collections.emptySet());
+        node.setProcessingExceptionHandler(processingExceptionHandler);
+
+        final InternalProcessorContext<Object, Object> internalProcessorContext = mock(InternalProcessorContext.class);
```

Comment:
Why do you not use `mockInternalProcessorContext()` here?

Reply:
Fixed